java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -44,17 +47,27 @@
   public void visitNode(Tree tree) {
     ClassTree classTree = (ClassTree) tree;
     for (Tree member : classTree.members()) {
-      if (member.is(Tree.Kind.VARIABLE) && hasNoVisibilityModifier((VariableTree) member)) {
+      if (isConsentWithCheck(member)) {
         IdentifierTree simpleName = ((VariableTree) member).simpleName();
         reportIssue(simpleName, "Explicitly declare the visibility for \"" + simpleName.name() + "\".");
       }
     }
   }
 
+  private static boolean isConsentWithCheck(Tree member) {
+    return member.is(Tree.Kind.VARIABLE)
+      && hasNoVisibilityModifier((VariableTree) member)
+      && isNotVisibleForTesting((VariableTree) member);
+  }
+
   private static boolean hasNoVisibilityModifier(VariableTree member) {
     ModifiersTree modifiers = member.modifiers();
     return !(ModifiersUtils.hasModifier(modifiers, Modifier.PUBLIC)
       || ModifiersUtils.hasModifier(modifiers, Modifier.PRIVATE)
       || ModifiersUtils.hasModifier(modifiers, Modifier.PROTECTED));
   }
+
+  private static boolean isNotVisibleForTesting(VariableTree member) {
+    return MODIFIERS_TREE_PREDICATE.test(member.modifiers());
```

Comment:
you can replace this call to your predicate by this: `!member.symbol().metadata().isAnnotatedWith("com.google.common.annotations.VisibleForTesting")`